pkgs/development/libraries/science/math/mkl/default.nix

```diff
@@ -87,10 +109,4 @@ stdenvNoCC.mkDerivation (rec {
     platforms = [ "x86_64-linux" "x86_64-darwin" ];
     maintainers = [ maintainers.bhipple ];
   };
- } // stdenvNoCC.lib.optionalAttrs stdenvNoCC.isLinux {
-  # Since on Linux binaries are unmodified, we can make them
-  # fixed-output derivations.
-	outputHashAlgo = "sha256";
-	outputHashMode = "recursive";
-	outputHash = "101krzh2mjbfx8kvxim2zphdvgg7iijhbf9xdz3ad3ncgybxbdvw";
```

Comment:
Can we just update this hash with pkgconfig included? It should still be the same thing each time.

Reply:
I am not sure how I get this. When I use `fakeSha256`, I just get the error:

```
error: path '/nix/store/kf5853mw8lgvhzhq4cm1y3p7k30f2b0a-mkl-2019.3.199' is not valid
```

Not the expected/actual hash you normally get with an incorrect hash for a fixed-output derivation. If I remove the lines that install the pkgconfig files, I do actually get the hash mismatch again.

As far as I understand, the difference is because this is not a fixed-output derivation anymore. The pkgconfig files contain the store path of the MKL library directory. Wouldn't these change with a change to `stdenvNoCC`?

Reply:
If this is correct, I'll change the derivation to make `mkl` a wrapper around the fixed-output derivation as @bhipple suggested.

Reply:
> Can we just update this hash with pkgconfig included? It should still be the same thing each time.

The pkg-config files reference `$out`, so as far as I know we can't have an up-front sha256.

Reply:
> That said, as mentioned before, I think we're probably OK to keep the derivation simple, since this isn't built on Hydra and since the src package saves us the network traffic of re-downloading it already.

Right, I wonder if it's worth complicating this more. MKL is not exactly a garden-variety library. Most users are probably in the data science/machine learning domain and have ample bandwidth anyway. And then it's nicer to have a derivation that is easy to understand and override.

I did notice, I still have a stray `echo` in the installPhase, which I'll remove.

Reply:
Agreed, and as noted above the `src` package is still a fixed-output derivation, so there's no wasted traffic/bandwidth externally downloading from Intel, and with an optimized nix store package rebuilds should be hard-linked anyways.